src/uct/tcp/tcp_ep.c

```diff
@@ -72,43 +89,19 @@ static inline void uct_tcp_ep_ctx_rewind(uct_tcp_ep_ctx_t *ctx)
     ctx->length = 0;
 }
 
-static ucs_status_t uct_tcp_ep_ctx_init(uct_tcp_iface_t *iface,
-                                        uct_tcp_ep_ctx_t *ctx)
+static void uct_tcp_ep_ctx_init(uct_tcp_ep_ctx_t *ctx)
 {
-    ucs_assertv(ctx->buf == NULL, "ctx=%p", ctx);
-
-    ctx->buf = ucs_malloc(ucs_max(iface->config.buf_size,
-                                  iface->config.short_size), "tcp_buf");
-    if (ctx->buf == NULL) {
-        return UCS_ERR_NO_MEMORY;
-    }
-
+    ctx->buf = NULL;
     uct_tcp_ep_ctx_rewind(ctx);
-
-    return UCS_OK;
 }
 
-static void uct_tcp_ep_ctx_cleanup(uct_tcp_ep_ctx_t *ctx)
+static inline void uct_tcp_ep_ctx_cleanup(uct_tcp_ep_ctx_t *ctx)
```

Comment:
minor: we can rename to 'uct_tcp_ep_ctx_reset', to indicate it's not called only during cleanup phase, but also during runtime